src/mongoose/createVersion.ts

```diff
@@ -0,0 +1,28 @@
+import type { MongooseAdapter } from '.';
+import type { CreateVersion } from '../database/types';
+import type { Document } from '../types';
+
+export const createVersion: CreateVersion = async function createVersion(this: MongooseAdapter,
+  { collectionSlug, parent, versionData, autosave, createdAt, updatedAt }) {
+  const VersionModel = this.versions[collectionSlug];
+
+
+  const doc = await VersionModel.create({
+    parent,
+    version: versionData,
+    autosave,
+    createdAt,
+    updatedAt,
+  });
+
+  let result: Document = doc.toJSON({ virtuals: true });
```

Comment:
might be able to just run the `create` operation as `lean()` so we don't need to do `doc.toJSON` afterward. nice to have i guess